lua/nvim-navic/init.lua

```diff
@@ -47,11 +47,49 @@ setmetatable(config.icons, {
 	end,
 })
 
+local function setup_nvim_navic(opts)
+	local function client_priority(client_name)
```

Comment:
Instead of having priority system with numbers, add the `prefernce` option like I did in nvim-navbuddy, (or maybe `priority` would be better name 🤔 ). Its just a list of lsp names, the earlier the name appears in the list, higher its priority. 
For example if preference is set to something like { `clangd`,  `tsserver` , `html` } and for some reason both tsserver and html are available for buffer, then tsserver will be selected since it comes earlier in the list.